yt_dlp/extractor/theplatform.py

```diff
@@ -297,6 +300,27 @@ def _real_extract(self, url):
 
         formats, subtitles = self._extract_theplatform_smil(smil_url, video_id)
 
+        note = 'Downloading m3u8 information'
+        errnote = 'Failed to download m3u8 information'
+        # on at least one site reading just the SMIL data misses some formats as well as subtitles.
+        # Therefore also read the m3u8 file and add those formats and subtitles
+        m3u8_url = url.split("&")[0]
```

Comment:
Truncating the query string after the first param seems pretty blunt and inexact

Can we narrow down the conditions when trying to request the url as m3u8 is necessary? Is it a specific URL format? Query params? Can you show an example URL that this works with (e.g. a url result from CBC)?

Reply:
> Can we narrow down the conditions when trying to request the url as m3u8 is necessary? Is it a specific URL format?
I've been trying to figure this out, but I haven't gotten very far. I would probably have to dive into the CBC Caffeine player code (>7k lines of code when formatted by beautifier.io) to get more details, and even then it's no guarantee. The only steadfast rule I've found: that blunt truncating can help - sometimes. Therefore the issue #7482 .
> Can you show an example URL that this works with (e.g. a url result from CBC)?
[http://www.cbc.ca/player/play/2233316419710](url), which resolves to [http://link.theplatform.com/s/ExhSPC/media/guid/2655402169/2233316419710?mbr=true&formats=MPEG4,FLV,MP3](url) smuggle_url with `'force_smil_url': True`.